src/modules/users/ActiveDirectoryJob.cpp

```diff
@@ -0,0 +1,91 @@
+/* === This file is part of Calamares - <https://calamares.io> ===
+ *
+ *   SPDX-FileCopyrightText: 2024 Simon Quigley <tsimonq2@ubuntu.com>
+ *   SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+#include "ActiveDirectoryJob.h"
+
+#include "Config.h"
+
+#include "GlobalStorage.h"
+#include "JobQueue.h"
+#include "utils/Logger.h"
+#include "utils/Permissions.h"
+#include "utils/System.h"
+
+#include <QDateTime>
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QTextStream>
+#include <QProcess>
+
+ActiveDirectoryJob::ActiveDirectoryJob(QStringList& activeDirectoryInfo)
+    : Calamares::Job()
+    , m_activeDirectoryInfo(activeDirectoryInfo)
+{
+}
+
+QString
+ActiveDirectoryJob::prettyName() const
+{
+    return tr( "Enroll system in Active Directory" );
+}
+
+QString
+ActiveDirectoryJob::prettyDescription() const
+{
+    return tr( "Enroll system in Active Directory" );
+}
+
+QString
+ActiveDirectoryJob::prettyStatusMessage() const
+{
+    return tr( "Enrolling system in Active Directory" );
+}
+
+Calamares::JobResult
+ActiveDirectoryJob::exec()
+{
+    QString username = m_activeDirectoryInfo.value(0);
+    QString password = m_activeDirectoryInfo.value(1);
+    QString domain = m_activeDirectoryInfo.value(2);
+    QString ip = m_activeDirectoryInfo.value(3);
+
+    Calamares::GlobalStorage* gs = Calamares::JobQueue::instance()->globalStorage();
+    QString rootMountPoint = gs ? gs->value("rootMountPoint").toString() : QString();
+
+    if (!ip.isEmpty()) {
+        QString hostsFilePath = !rootMountPoint.isEmpty() ? rootMountPoint + "/etc/hosts" : "/etc/hosts";
```

Comment:
It doesn't really matter if `rootMountPoint` is empty (if it is, `rootMountPoint + "something"` is "something"). So this can simplify to `const QString hostsFilePath = rootMountPoint + "/etc/hosts";` or even simpler, `Calamares::System::instance()->targetPath(QStringLiteral("/etc/hosts"));` which is the API for getting paths to the target system.

Reply:
"even simpler" is relative, of course :/